src/TabComplete.js

```diff
@@ -45,21 +48,32 @@ class TabComplete {
         this.isFirstWord = false; // true if you tab-complete on the first word
         this.enterTabCompleteTimerId = null;
         this.inPassiveMode = false;
+        this.memberTabOrder = {};
+        this.memberOrderSeq = 0;
     }
 
     /**
-     * @param {Entry[]} completeList
+     * Call this when a a UI element representing a tab complete entry has been clicked
+     * @param {entry} The entry that was clicked
      */
-    setCompletionList(completeList) {
-        this.list = completeList;
+    onEntryClick(entry) {
         if (this.opts.onClickCompletes) {
-            // assign onClick listeners for each entry to complete the text
-            this.list.forEach((l) => {
-                l.onClick = () => {
-                    this.completeTo(l);
-                }
-            });
+            this.completeTo(entry);
+        }
+    }
+
+    loadEntries(room) {
```

Comment:
So you've changed the public API of `TabComplete` to take a `Room` rather than `Entry[]`, meaning it cannot be reused across multiple components if they want different behaviours - we're baking in `RoomView` behaviour.

This may turn out to be fine - but I'm worried that there will be a number of components which would want to use the functionality which `TabComplete` provides - e.g. partially typing a user into the invite box then tab completing to the user to invite, ditto with searching rooms. This is why `TabComplete` was designed to accept a set of `Entry` objects rather than bake in `Room` specific semantics, and is why some tab-complete logic existed in `RoomView`. This made it really easy to extend when @ara4n wanted to include `RoomMembers` in addition to `/commands` which is what this was originally designed to do.